File: beamng/msg/vehicle_control_stamped.py
from pycdr2 import IdlStruct
from pycdr2.types import float32
from dataclasses import dataclass

from msg.builtin_interfaces import Time

@dataclass
class VehicleControlStamped(IdlStruct, typename="VehicleControlStamped"):
  stamp: Time
  throttle: float32
  brake: float32
  steering: float32

File: zenoh_bridge/src/msg/vehicle_control_stamped.rs
use serde::{Deserialize, Serialize};
use zenoh_ros_type::{builtin_interfaces};

#[derive(Serialize, Deserialize, PartialEq, Clone)]
pub struct VehicleControl {
    pub stamp: builtin_interfaces::Time,
    pub throttle: f32,
    pub brake: f32,
    pub steering: f32,
}